### ISSUE TITLE
[<Ray component: Core|RLlib|etc...>] 

### ISSUE THREAD

**Issue by BaitingLuo** [ray-project/ray#32228]
### What happened + What you expected to happen

By running the given example of AlphaZero, line 51: mask = obs["action_mask"] in ray/rllib/algorithms/alpha_zero/ranked_rewards.py" reports there is no "action_mask" in obs dictionary.

(RolloutWorker pid=13910) IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
^C(AlphaZero pid=14070) 2023-02-05 22:29:31,974 WARNING algorithm_config.py:489 -- Cannot create AlphaZeroConfig from given `config_dict`! Property __stdout_file__ not supported.
(AlphaZero pid=14070) 2023-02-05 22:29:31,976   INFO algorithm.py:504 -- Current log_level is WARN. For more information, set 'log_level': 'INFO' / 'DEBUG' or use the -v and -vv flags.

### Versions / Dependencies

Ray version 2.2.0, python version 3.7, OS ubuntu 18.04.

### Reproduction script

from ray.rllib.algorithms.alpha_zero import AlphaZeroConfig
from ray import air
from ray import tune
config = AlphaZeroConfig()
# Print out some default values.
print(config.shuffle_sequences)
# Update the config object.
config.training(lr=tune.grid_search([0.001, 0.0001]))
# Set the config object's env.
config.environment(env="CartPole-v1")
# Use to_dict() to get the old-style python config dict
# when running with tune.
tune.Tuner(
    "AlphaZero",
    run_config=air.RunConfig(stop={"episode_reward_mean": 200}),
    param_space=config.to_dict(),
).fit()

### Issue Severity

High: It blocks me from completing my task.

**Comment by ArturNiederfahrenhorst** [comment]
Hi @BaitingLuo ,
Your environment does not satisfy the AlphaZero's needs.
Essentially, you can not simply throw any environment at it, but only one that implements `get_state` and returns observations with action masks.
Have a look at our example: https://github.com/ray-project/ray/blob/095960cf650bcbd672e72b6e8e383a1812ce0ab9/rllib/examples/env/cartpole_sparse_rewards.py#L8
The example is also a cartpole example so you might be able to simply modify it slightly.

**Comment by ArturNiederfahrenhorst** [comment]
I'm counting this as an enhancement, since we might want to simply offer more informative errors here.
There is no plan for "auto convertig" environments into AlphaZero though.

**Comment by Alian3785** [comment]
I agree, there should at least be more comprehensive error message or clear point in documentation that you can`t just plug an algorithm and it learns, as with almost all other algorithms. And that you have to modify an env.

Also please get rid of extra dots in documentation example. They cause SyntaxError
![extra dots error](https://github.com/ray-project/ray/assets/60827442/3a1a49bb-4f3d-4a55-b2fb-7a7fd27822b1)
